packages/payload/src/fields/hooks/afterChange/promise.ts

```diff
@@ -105,7 +105,8 @@ export const promise = async ({
               global,
               operation,
               previousDoc,
-              previousSiblingDoc: previousDoc?.[field.name]?.[i] || ({} as Record<string, unknown>),
+              previousSiblingDoc:
+                previousSiblingDoc?.[field.name]?.[i] || ({} as Record<string, unknown>),
```

Comment:
Is this expected?
Imagine a user reorders items in an array and then resaves it?
The position in the array has changed so the previous data just won't line up correctly for the hook to work as expected.

Shouldn't we `find` by the `id` (internal) to the array so that we can have the hook's previous value be more easily used?

Reply:
I actually had that case in my project and it's kinda confusing, yes. In my case I then used the `afterChange` hook on the array field itself to do the comparison manually.

Does an array item always have an (internal) `id` in payload?

For example, would this ever be possible:

```ts
{
  array: [
    "item1",
    "item2",
    "item3"
  ]
}
```

or is an array item always an object with at least an `id` property:

```ts
{
  array: [
    { id: "id1", text: "item1" }
    { id: "id2", text: "item1" }
    { id: "id3", text: "item1" }
  ]
}
```

If yes, I absolutely agree and will change the code accordingly.

Reply:
Yes, by the time the operations get to the `afterChange` hook, you can expect the items to always have an `id`. I believe this happens in the `beforeChange` hook.

Also, we have a function already called `getExistingRowDoc` which might be what you need to use.

Reply:
@DanRibbens i just pushed another commit to address this issue.